test/CMakeLists.txt

```diff
@@ -1,9 +1,24 @@
 cmake_minimum_required(VERSION 3.7)
 
 enable_testing()
-find_package(GTest REQUIRED)
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  # Specify the commit you depend on and update it regularly.
+  URL https://github.com/google/googletest/archive/refs/tags/v1.13.0.zip
+  # FIND_PACKAGE_ARGS NAMES GTest
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
```

Comment:
That's cool but I think there should be an option to use system provided GTest still.
Sometimes, projects that deal with networking/security (such as this one) are built in airgapped environments.

Reply:
"Specify the commit you depend on and update it regularly." might be fine for GTest, in general we prefer to depend on distro (Debian in most cases) libraries that get some security team support (and yet are stable in version). Not everywhere it's possible (npm, python packages), but it's a pain to "update commits regularly".

For gtest I would not care, but still, it would be best if it worked internally without calling gh - internal mirror on bitbucket in this case. So URL would have to be a variable (?). And then we have a mirror to update too. We try to care about the delivery chain.